ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/lexer/LexerDirectoryProcessor.java

```diff
@@ -0,0 +1,34 @@
+package edu.usb.argos.ASTProcessor.lexer;
+
+import org.antlr.v4.runtime.CommonTokenStream;
+
+import java.io.IOException;
+import java.nio.file.*;
+import java.util.HashMap;
+import java.util.Map;
+
+public class LexerDirectoryProcessor {
+    private final LexerFileProcessor fileProcessor;
+
+    public LexerDirectoryProcessor() {
+        this.fileProcessor = new LexerFileProcessor();
+    }
+
+    public Map<String, CommonTokenStream> getTokensFromDirectoryByFile(String directoryPath) throws IOException {
+        Map<String, CommonTokenStream> tokensByFile = new HashMap<>();
+
+        Files.walk(Paths.get(directoryPath))
```

Comment:
Add a try-catch in Files.walk

Reply:
I added a try with resource, to delete the warning in Files.walk